### ISSUE TITLE
i18n package for zog

### ISSUE THREAD

**Issue by Oudwins** [Oudwins/zog#26]
The idea is to go one step futher in supporting i18n and have an actual small package dedicated to this. It will have a function that takes a language map and will server default errors based on that map.

Currently it is only planned to support the translation to spanish

**Comment by Oudwins** [comment]
@cachesdev has agreed to help handle some of the translations!

**Comment by cachesdev** [comment]
I was thinking about this, and there are many ways we can proceed, you can always go with a direct translation where the types are translated. in example, a string will become "cadena" a number will become "numero" and so on, or you could mix both. the third option is to say "field" or "campo" instead of the type itself, but this assumes validation for a form.

```go
// translate types Cadena
p.ErrCodeMin:  "la cadena debe contener al menos {{min}} caracter(es)",

// Original type
p.ErrCodeMin:  "string debe contener al menos {{min}} caracter(es)",

// Refer to it as a field
p.ErrCodeMin:  "el campo debe contener al menos {{min}} caracter(es)",
```

in Latam we usually refer to the translated version of types, not the original english terms. in a conversation it would be normal to talk about "cadenas, booleanos, mapas y listas" so the direct translation has some merit.

**Comment by Oudwins** [comment]
I think you are right about direct translation being the right approach. Ideally I would like for this errors to be fine to display to non technical users. 

The reason field is not being used and instead the type name is being used is because: 
1. That is how zod does it 
2. I don't want the validation messages to only be valid for user input

**Comment by Oudwins** [comment]
I have the start of a working version for the i18n package. I'll hopefully push it up today and you can take a look at it.

The main idea is:

1. Have a simple i18n package which has a single function to set the languages you are going to use & their language maps plus the default language
2. Have sub packages with error maps for each language we want to support. For example `zog/i18n/es` would be the package for spanish. 

This way if you need to support spanish and english you would do something like this:

```go
zi18n.SetLanguagesMap(map[string]conf.LangMap["es": zes.Map, "en": zen.Map,], "en") // default lang = en
```

Then when you parse:

```go
errs := schema.Parse(data, &dest, z.withCtxValue("lang", "es"))
```

**Comment by Oudwins** [comment]
@cachesdev Its ready. You can see the PR at #28

**Comment by cachesdev** [comment]
that looks good. so by the looks of it a language package will only export a variable `Map` of type `LangMap`? do you want me to open a PR with the `zes` package on that branch?

**Comment by Oudwins** [comment]
I also still need to add documentation and move the en lang map before mergin that PR

**Comment by cachesdev** [comment]
@Oudwins opened the PR

**Comment by Oudwins** [comment]
RESOLVED IN #28
